lib/libxdp/libxdp.c

```diff
@@ -1531,7 +1565,8 @@ static int xdp_multiprog__link_pinned_progs(struct xdp_multiprog *mp)
 	goto out;
 }
 
-static int xdp_multiprog__fill_from_fd(struct xdp_multiprog *mp, int prog_fd)
+static int xdp_multiprog__fill_from_fd(struct xdp_multiprog *mp, int prog_fd,
+				       bool fill_hw)
```

Comment:
Instead of a bool, how about turning this into an 'int hw_fd' and handle that at the bottom after the main programs are added...